src/snowflake/cli/api/utils/definition_rendering.py

```diff
@@ -340,6 +345,17 @@ def render_definition_template(
         project_context[CONTEXT_KEY]["env"] = environment_overrides
         return ProjectProperties(project_definition, project_context)
 
+    # need to have the metrics added here since we add defaults to the
+    # definition that the user might not have added themselves later
+    metrics = get_cli_context().metrics
+
+    # render_definition_template is invoked multiple times both by the user,
+    # and by us so we should make sure we don't overwrite a 1 with a 0 here
+    metrics.add_counter(CLICounterField.PDF_TEMPLATES, 0)
+
+    if _has_referenced_vars_in_definition(template_env, definition):
+        metrics.set_counter(CLICounterField, 1)
+
```

Comment:
I meant to factor this whole thing into a function to not distract the reader from the code itself.